### PR TITLE
Quiz result page

### PR DESCRIPTION
**Pull request by Digicat13**
Created page for completing quizzes

- test-info
At first user sees info about quiz (time limits ans available attepmts) and name input if it wasnt specified.  When user press "Start Quiz", the stopwatch is starting and user see first question instead of testinfo

- quiz-header
Contains testname, proggress of quiz (#currentQuestion/QuestionsCount), stopwatch to calculate duration of testing, timeout for quizTimeLimit or testTimeLimit (if any were defined), and events to handle timeouts (if question time ends it emits to go to next question, for testTimeLimit to submit the quiz).

- test-question
Contains card with question and available answers. If number of correct answers is 1 then options are with radiobutons, else there are checkboxes.

- quiz-page 
Parent component which contains test-info, quiz-header  and questions. After submiting the test user will be redirected to quiz-result page.

- quiz-result 
Contains user score and all questions with highlighted correct answers and selected wrong answers.

### CODE REVIEW

File: src/app/pages/quiz-page/quiz-page/quiz-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      this.testingId = params.get('id');
+    });
+  }
+
+  async ngOnInit(): Promise<void> {
+    await this.getTesting(this.testingId);
+    this.getTest(this.testing?.testId);
+  }
+
+  async getTesting(testingId: string): Promise<void> {
```
Comment (olenarehushleobit) [inline src/app/pages/quiz-page/quiz-page/quiz-page.component.ts]: Please change `Promise `to `Observable`

File: src/app/pages/quiz-page/quiz-page/quiz-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    await this.testingService
+      .getTesting(testingId)
+      .toPromise()
```
Comment (olenarehushleobit) [inline src/app/pages/quiz-page/quiz-page/quiz-page.component.ts]: use `subscribe `insread of `toPromise`

Comment (Digicat13) [reply src/app/pages/quiz-page/quiz-page/quiz-page.component.ts]: Changed

File: src/app/pages/quiz-page/quiz-page/quiz-page.component.ts (hunk 1 above)
Comment (Digicat13) [reply src/app/pages/quiz-page/quiz-page/quiz-page.component.ts]: Changed

File: src/app/pages/quiz-page/quiz-page/quiz-page.component.ts (hunk 2 above)
Comment (Digicat13) [reply src/app/pages/quiz-page/quiz-page/quiz-page.component.ts]: Changed to mergeMap

**Review by olenarehushleobit — APPROVED** [review]

**Pull request merged**